### ISSUE TITLE
[Bug]: Window contents disappear when webcontents is focused after a navigation all while the window is hidden

### ISSUE THREAD

**Issue by pushkin-** [electron/electron#28255]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

12.0.1, 11.2.1

### What operating system are you using?

Windows

### Operating System Version

Windows 10 v1909 and 20H2

### What arch are you using?

x64

### Last Known Working Electron version

9?

### Expected Behavior

When the BrowserWindow does a navigation while the computer is locked, unlocking the machine shows the same window as before

### Actual Behavior

The window is blank. The contents reappear once I trigger a repaint by moving the window

1. start the app
2. in the devtools, execute `setTimeout(() => location = location.href, 2000)`
3. bring a maximized (maybe not necessary) application over the Electron app and wait 2 seconds until the navigation happens
4. bring the Electron app over the application by selecting it in the taskbar (but don't hover over the taskbar to show the preview image Windows will display). Notice the page is black.

This worked fine in Electron 9 I believe.

Removing my `focus` call fixes it. In my actual app I have a `dom-ready` handler in place that focuses the webcontents to work around [this issue](https://github.com/electron/electron/issues/11735).

```
(async () => {
	const { app, BrowserWindow } = require("electron");
	app.on("window-all-closed", async () => {
		app.quit();
	});

	await app.whenReady();
	const mainWindow = new BrowserWindow();

	mainWindow.webContents.on("did-navigate", () => {
                // In my actual app, I am using a BrowserView and a dom-ready handler to work around issue https://github.com/electron/electron/issues/11735
		mainWindow.webContents.focus()
	});

	await mainWindow.loadURL("https://google.com");
	mainWindow.webContents.openDevTools();
})();
```

**Comment by pushkin-** [comment]
[related issue](https://github.com/electron/electron/issues/27353)

**Comment by pushkin-** [comment]
Seems like disabling background throttling addresses this, though I don't want to do that in my app

**Comment by pushkin-** [comment]
This isn't actually specific to locking the machine. This repros if another app is covering the Electron app.

**Comment by nornagon** [comment]
@pushkin- the repro instructions here begin with "start the app" but don't mention what app. Is there a fiddle or other app that is missing here?

**Comment by pushkin-** [comment]
@nornagon the block of code in the post is meant to be main.js, which is the app to start.

**Comment by nornagon** [comment]
Interestingly this DCHECKs on current master:

```
[9351:0427/110930.720402:FATAL:render_frame_host_impl.cc(10421)] Check failed: browser_should_replace_current_entry == params.should_replace_current_entry (1 vs. 0)
0   Electron Framework                  0x0000000118585009 base::debug::CollectStackTrace(void**, unsigned long) + 9
1   Electron Framework                  0x00000001184a12b3 base::debug::StackTrace::StackTrace() + 19
2   Electron Framework                  0x00000001184b9d3f logging::LogMessage::~LogMessage() + 175
3   Electron Framework                  0x00000001184bac3e logging::LogMessage::~LogMessage() + 14
4   Electron Framework                  0x00000001175b8fb8 content::RenderFrameHostImpl::VerifyThatBrowserAndRendererCalculatedDidCommitParamsMatch(content::NavigationRequest*, content::mojom::DidCommitProvisionalLoadParams const&, mojo::InlinedStructPtr<content::mojom::DidCommitSameDocumentNavigationParams>) + 9352
5   Electron Framework                  0x000000011759a999 content::RenderFrameHostImpl::DidCommitNavigationInternal(std::__1::unique_ptr<content::NavigationRequest, std::__1::default_delete<content::NavigationRequest> >, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::InlinedStructPtr<content::mojom::DidCommitSameDocumentNavigationParams>) + 2809
6   Electron Framework                  0x00000001175999eb content::RenderFrameHostImpl::DidCommitNavigation(content::NavigationRequest*, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>) + 1339
7   Electron Framework                  0x00000001175ced92 base::internal::Invoker<base::internal::BindState<void (content::RenderFrameHostImpl::*)(content::NavigationRequest*, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>), base::internal::UnretainedWrapper<content::RenderFrameHostImpl>, content::NavigationRequest*>, void (mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>)>::RunOnce(base::internal::BindStateBase*, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>&&, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>&&) + 82
8   Electron Framework                  0x00000001152912fe content::mojom::NavigationClient_CommitNavigation_ForwardToCallback::Accept(mojo::Message*) + 318
9   Electron Framework                  0x000000011894eaa4 mojo::InterfaceEndpointClient::HandleValidatedMessage(mojo::Message*) + 1668
10  Electron Framework                  0x00000001189542ef mojo::MessageDispatcher::Accept(mojo::Message*) + 271
11  Electron Framework                  0x0000000118950237 mojo::InterfaceEndpointClient::HandleIncomingMessage(mojo::Message*) + 55
12  Electron Framework                  0x0000000118ce3f74 IPC::(anonymous namespace)::ChannelAssociatedGroupController::AcceptOnProxyThread(mojo::Message) + 756
13  Electron Framework                  0x0000000118ce0c4c base::internal::Invoker<base::internal::BindState<void (IPC::(anonymous namespace)::ChannelAssociatedGroupController::*)(mojo::Message), scoped_refptr<IPC::(anonymous namespace)::ChannelAssociatedGroupController>, mojo::Message>, void ()>::RunOnce(base::internal::BindStateBase*) + 140
14  Electron Framework                  0x000000011851c110 base::TaskAnnotator::RunTask(char const*, base::PendingTask*) + 400
15  Electron Framework                  0x000000011854089b base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWorkImpl(base::sequence_manager::LazyNow*) + 1083
16  Electron Framework                  0x000000011854006d base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork() + 93
17  Electron Framework                  0x000000011859bf51 base::MessagePumpCFRunLoopBase::RunWork() + 65
18  Electron Framework                  0x0000000118594b42 base::mac::CallWithEHFrame(void () block_pointer) + 10
19  Electron Framework                  0x000000011859b97f base::MessagePumpCFRunLoopBase::RunWorkSource(void*) + 63
20  CoreFoundation                      0x00007fff2fe36d52 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
21  CoreFoundation                      0x00007fff2fe36cf1 __CFRunLoopDoSource0 + 103
22  CoreFoundation                      0x00007fff2fe36b0b __CFRunLoopDoSources0 + 209
23  CoreFoundation                      0x00007fff2fe3583a __CFRunLoopRun + 927
24  CoreFoundation                      0x00007fff2fe34e3e CFRunLoopRunSpecific + 462
25  HIToolbox                           0x00007fff2ea61abd RunCurrentEventLoopInMode + 292
26  HIToolbox                           0x00007fff2ea617d5 ReceiveNextEventCommon + 584
27  HIToolbox                           0x00007fff2ea61579 _BlockUntilNextEventMatchingListInModeWithFilter + 64
28  AppKit                              0x00007fff2d0a7039 _DPSNextEvent + 883
29  AppKit                              0x00007fff2d0a5880 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1352
30  AppKit                              0x00007fff2d09758e -[NSApplication run] + 658
31  Electron Framework                  0x000000011859cb2c base::MessagePumpNSApplication::DoRun(base::MessagePump::Delegate*) + 348
32  Electron Framework                  0x000000011859b32c base::MessagePumpCFRunLoopBase::Run(base::MessagePump::Delegate*) + 140
33  Electron Framework                  0x000000011854144b base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run(bool, base::TimeDelta) + 651
34  Electron Framework                  0x00000001184f52da base::RunLoop::Run(base::Location const&) + 890
35  Electron Framework                  0x00000001170c503f content::BrowserMainLoop::RunMainMessageLoop() + 111
36  Electron Framework                  0x00000001170c6da2 content::BrowserMainRunnerImpl::Run() + 82
37  Electron Framework                  0x00000001170c217b content::BrowserMain(content::MainFunctionParams const&) + 251
38  Electron Framework                  0x000000011574f7e5 content::ContentMainRunnerImpl::RunBrowser(content::MainFunctionParams&, bool) + 1397
39  Electron Framework                  0x000000011574f201 content::ContentMainRunnerImpl::Run(bool) + 529
40  Electron Framework                  0x000000011574e0e3 content::RunContentProcess(content::ContentMainParams const&, content::ContentMainRunner*) + 2675
41  Electron Framework                  0x000000011574e1dc content::ContentMain(content::ContentMainParams const&) + 44
42  Electron Framework                  0x0000000113e12357 ElectronMain + 135
43  Electron                            0x00000001089a0faa main + 298
44  libdyld.dylib                       0x00007fff69ed6cc9 start + 1
45  ???                                 0x0000000000000006 0x0 + 6
```

and on the latest nightly, it crashes with:

```
Thread 36 (crashed)
 0  Electron Framework!device::GeolocationSystemPermissionManager::GetObserverList() [scoped_refptr.h : 197 + 0x0]
    rax = 0x0000700011862970   rdx = 0x0000000000000cdc
    rcx = 0x0000000000000000   rbx = 0x0000700011862970
    rsi = 0x0000000000000000   rdi = 0x0000700011862970
    rbp = 0x0000700011862910   rsp = 0x0000700011862910
     r8 = 0x0000000000001bfa    r9 = 0x000000000000cc3d
    r10 = 0x00000000ffffc7ff   r11 = 0x0000000000000000
    r12 = 0x00007f9cb2dce7e8   r13 = 0x00007f9cb2dce760
    r14 = 0x0000000000000000   r15 = 0x0000700011862968
    rip = 0x000000011396bcb7
    Found by: given as instruction pointer in context
 1  Electron Framework!device::NetworkLocationProvider::NetworkLocationProvider(scoped_refptr<network::SharedURLLoaderFactory>, device::GeolocationSystemPermissionManager*, scoped_refptr<base::SingleThreadTaskRunner>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, device::PositionCache*) [network_location_provider.cc : 64 + 0xf]
    rbp = 0x00007000118629d0   rsp = 0x0000700011862920
    rip = 0x0000000111999acb
    Found by: previous frame's frame pointer
 2  Electron Framework!device::LocationArbitrator::NewNetworkLocationProvider(scoped_refptr<network::SharedURLLoaderFactory>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) [memory : 2006 + 0x11]
    rbp = 0x0000700011862a20   rsp = 0x00007000118629e0
    rip = 0x0000000111996ace
    Found by: previous frame's frame pointer
 3  Electron Framework!device::LocationArbitrator::RegisterProviders() [location_arbitrator.cc : 122 + 0x13]
    rbp = 0x0000700011862a70   rsp = 0x0000700011862a30
    rip = 0x000000010e53548c
    Found by: previous frame's frame pointer
 4  Electron Framework!device::LocationArbitrator::StartProvider(bool) [location_arbitrator.cc : 65 + 0x8]
    rbp = 0x0000700011862a90   rsp = 0x0000700011862a80
    rip = 0x0000000111996796
    Found by: previous frame's frame pointer
[...]
```

... but the above are unrelated.

I can reproduce this on 12.0.0 on Windows 10 in a VM. In that setup the window is white rather than black when switching back to it, but I suspect it's the same underlying cause. The window redraws when resized. We must be missing some sort of message to the renderer process to tell it to redraw itself.

**Comment by nornagon** [comment]
Here's a fiddle for easy reproing: https://gist.github.com/b189dcb2109b60cd79c8eef6c6a60aee

**Comment by pushkin-** [comment]
@nornagon Cool - thanks! And in my app, I set my background color to blackish, so I suspect the white/black difference doesn't matter as you mentioned

**Comment by pushkin-** [comment]
still repros in Electron 16.0.5

**Comment by pushkin-** [comment]
can't repro anymore. Seems like it got fixed around 17.3.0?
